### ISSUE TITLE
Add Support for HeroLab Statblocks

### ISSUE THREAD

**Issue by Lavaeolous** [Lavaeolous/PF1-StatBlock-Converter-Module#461]
[Gargoyle test CR 4.txt](https://github.com/Lavaeolous/PF1-StatBlock-Converter-Module/files/6582757/Gargoyle.test.CR.4.txt)
[HeroLab_PC.txt](https://github.com/Lavaeolous/PF1-StatBlock-Converter-Module/files/6582758/HeroLab_PC.txt)

**Comment by Hanhula** [comment]
I've got a variety of HL statblocks on hand if you want more of these to use for testing, all of varying levels with PC, NPC, and monster details (often including mythic) - let me know if you'd like any!

**Comment by karpana** [comment]
My observartion with Herolab (classic) stat blocks, is that it's the "line of dashes" after "Special Abilities"... that seems to prevent import.  
This was obserrved/validated on 0.8.6, with the latest version of SBC on June 5th.

**Comment by karpana** [comment]
The other things I do to Herolab (Classic) stat blocks includes

- remove the "creature description", if it is present
- remove the line of dashes before the creature's name
- remove the "line of dashes" after Special Abilities
- remove the herolab Trademarks lines (last 2 at the bottom)

When i do this, I get enough of the stat block imported that I can at least work with it in Foundry

**Comment by Bobsson** [comment]
In addition to the above changes, I also have to delete line breaks in the attacks.  For example
```
Melee +1 heavy shield bash +23 (1d8+7) or   
+2 khopesh +23/+18/+13 (1d8+15/17-20) or   
slam +15 (1d8+4)
```
will not work, but 
```
Melee +1 heavy shield bash +23 (1d8+7) or   +2 khopesh +23/+18/+13 (1d8+15/17-20) or   slam +15 (1d8+4)
``` 
will.

**Comment by adventuremagic123** [comment]
I'm so glad to see others using Hero Lab stat blocks.  This is the best way to use this module because it's the only way to get consistent stat blocks in a predictable format.

I helped test Hero Lab stat blocks 2 years ago.  With all the changes to FVTT and the PF1e game system, there's new breakages.

**Comment by adventuremagic123** [comment]
I can't use FVTT without an importer for NPCs -- too much work.  Hopefully, these issues will eventually be addressed, but until then MapTool offers a good alternative (or you could try another FVTT game system).

I'm currently using MapTool with a framework (game system) I created for PF1e.  Currently, it works best with MapTool 1.10.4 due to issues in subsequent MapTool releases (hopefully to be fixed soon in 1.12.0).

You can my framework at:

https://github.com/adventuremagic123/PF1e-And-3.5-Framework-For-MapTool

I've used it with Hero Lab to build up my private library of thousands of NPCs and it's stable -- meaning it works with all my subsequent framework releases because I use the command software design pattern.

You can get MapTool from:

https://github.com/RPTools/maptool/releases/tag/1.10.4

Click "Assets" and download MapTool 1.10.4 from there.

I strongly recommend the Elevation Library because it will allow you to do your maps in 3D on MapTool, but it's an advanced topic where you should read about token libraries first:

https://github.com/melek/lib_elevation

FVTT is a really great virtual tabletop but it is still changing rapidly and it's hard (at least for me) to keep everything working because they apparently can't use S.O.L.I.D. techniques and I don't fault them for that.  Probably would be too heavy weight for FVTT to make progress.

**Comment by karpana** [comment]
@adventuremagic123 

I'll definitely have to check out the process and tools you're using. But I do have to say that migrating away from MT to FVTT made the whole experience of running PFRPG 1E simpler for me, and far less frustrating for my players. 

The compatibility and performance issues I experienced with MT made it difficult to keep up with the release cadence (v0.6 through v0.8x)

FVTT is still changing quickly, but the community is keeping up well. 

@Lavaeolous I'm about to prep another Pathfinder AP (jade Regent book 2l,  so I'll make a point to document which challenges I have with importing statblocks. 

I have found that using the community built stat block libraries are super helpful for the paizo APs, but I still struggle with non-paizo imports.
